src/pages/main/mainPage.js

```diff
@@ -0,0 +1,26 @@
+import { createContext, useContext, useReducer } from "react";
+import "./mainPage.css";
+import List from "./parts/list";
```

Comment:
I think, it's better to rename "parts" folder to "components". This name is more common in that case.